tests/test_simple_nws.py

```diff
@@ -313,3 +333,37 @@ async def test_nws_alerts_all_zones_second_alert(aiohttp_client, mock_urls):
     assert alerts
     assert new_alerts == alerts
     assert len(alerts) == 2
+
+
+async def test_retries(aiohttp_client, mock_urls, monkeypatch):
+    app = setup_app()
+    client = await aiohttp_client(app)
+    nws = SimpleNWS(*LATLON, USERID, client)
+    await nws.set_station(STATION)
+
+    mock_update = AsyncMock()
+    mock_update.side_effect = [aiohttp.ClientResponseError, None]
```

Comment:
ClientResponseError doesn't work here unless we provide it with a fully formed response object, etc. This seems like overkill.  Id like to figure out a way to get this test to work as it is a unit test for this functionality, including passing parameters correctly.  This isnt so obvious in async code without the test.

Reply:
aioresponses looks promising

Reply:
Tests now use server based errors for testing `ClientResponseError` based errors. Unit test now additionally mocks out the error check which allows us to raise any error in that test